### PR TITLE
Add nftables support

### PR DESCRIPTION
**Pull request by galal-hussein**
Signed-off-by: galal-hussein <hussein.galal.ahmed.11@gmail.com>

- https://github.com/k3s-io/k3s/issues/6649
- https://github.com/k3s-io/klipper-lb/issues/34

### CODE REVIEW

File: entry [hunk 1]
```diff
@@ -1,29 +1,83 @@
...
+    else 
+        mode=legacy
+    fi
+}
+
+set_nft() {
+    bin_dir="/sbin"
+
+    for i in iptables iptables-save iptables-restore ip6tables; do 
+        ln -sf "xtables-nft-multi" "$base_path/$i";
```
Comment (brandond) [inline entry:29]: Should this be `bin_dir` not `base_path`? I don't see `base_path` defined anywhere.
```suggestion
        ln -sf "xtables-nft-multi" "$bin_dir/$i"; 
```

File: entry [hunk 2]
```diff
@@ -1,29 +1,83 @@
...
+    lsmod | grep "nf_tables" 2> /dev/null
+    if [ $? = 0 ]; then
+        mode=nft
+    else 
+        mode=legacy
+    fi
+}
+
+set_nft() {
+    bin_dir="/sbin"
```
Comment (brandond) [inline entry:26]: Could probably define this up at the top instead of setting it to the same thing in both functions.

**Review by brandond — CHANGES_REQUESTED** [review]

File: entry (hunk 1 above)
Comment (galal-hussein) [reply entry:29]: yeah its a typo, fixing now

File: entry [hunk 3]
```diff
@@ -1,29 +1,80 @@
...
+
+set_nft() {
+    for i in iptables iptables-save iptables-restore ip6tables; do 
+        ln -sf "xtables-nft-multi" "$bin_path/$i"; 
+    done
+}
+
+set_legacy() {
+    for i in iptables iptables-save iptables-restore ip6tables; do 
+        ln -sf /sbin/xtables-nft-multi "$bin_dir/$i";
```
Comment (brandond) [inline entry:33]: I think you meant to set bin_dir at the top and make these both bin_dir?

**Review by brandond — CHANGES_REQUESTED** [review]
I would probably set bin_dir at the top of the file instead of right before calling the functions, just for readability's sake.

Also, one of the set functions still uses the wrong variable.

**Comment by galal-hussein** [comment]
> I would probably set bin_dir at the top of the file instead of right before calling the functions, just for readability's sake.
> 
> Also, one of the set functions still uses the wrong variable.

yeah sorry the file was saved incorrectly, I fixed it now

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**